vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java

```diff
@@ -234,30 +233,22 @@ private static void resolveSpelExpressions(Properties properties, boolean ignore
         }
     }
 
-    private static void processSystemProperties(Properties properties)
+    public static void reset()
     {
-        Iterator<Entry<Object, Object>> iterator = properties.entrySet().iterator();
-        while (iterator.hasNext())
-        {
-            Entry<Object, Object> entry = iterator.next();
-            String key = (String) entry.getKey();
-            if (key.startsWith(SYSTEM_PROPERTIES_PREFIX))
-            {
-                System.setProperty(StringUtils.removeStart(key, SYSTEM_PROPERTIES_PREFIX), (String) entry.getValue());
-                iterator.remove();
-            }
-        }
+        instance = null;
     }
 
-    public static void reset()
+    public String getUndecryptedProperty(String key)
     {
-        instance = null;
+        return properties.getProperty(key);
     }
 
-    public Properties getProperties()
+    public Properties getProperties(StringEncryptor stringEncryptor)
     {
         isReset();
-        return (Properties) properties.clone();
+        PropertiesDecryptor decryptor = new PropertiesDecryptor(stringEncryptor);
+        Properties processedProperties = SystemPropertiesProcessor.process(decryptor, properties);
```

Comment:
non-encrypted system properties are set twice with this approach

Reply:
the things are not changed much: the properties are not set twice, but they are still processed twice

Reply:
the proposed solution:
- find encryptor password or use a default one
- create StringEncryptor instance
- process system properties one (find by prefix, decrypt, set)